src/csharp/Grpc.Tools.CSharpGenerator/Program.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Grpc.Tools.CSharpGenerator
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            try
+            {
+                var projectDir = args[0];
+                var intermediateOutputDir = args[1];
+
+                var osName = string.Empty;
+
+                if (Grpc.Core.Internal.PlatformApis.IsWindows)
+                    osName = "windows_{0}";
+                else if (Grpc.Core.Internal.PlatformApis.IsMacOSX)
+                    osName = "macosx_{0}";
+                else if (Grpc.Core.Internal.PlatformApis.IsLinux)
+                    osName = "linux_{0}";
+                else
+                {
+                    Console.Error.WriteLine("Could not detect your operating system. The only supported operating systems are Windows, Mac OSX and Linux");
+                    return;
+                }
+
+                var bitness = string.Empty;
+                if (Grpc.Core.Internal.PlatformApis.Is64Bit)

```

Comment:
For windows, x86 and x64 protoc binaries are interchangeable, on linux though, the difference between 64 and 32bit matters (for mac osx it sometimes matters). So for completeness, we select between the 32 and 64 bit version always.

This is actually more important for gRPC C# native extension loading - where we load shared library of appropriate bitness depending on the bitness of the current process (but that's a separate story).